handlers.go

```diff
@@ -229,6 +241,31 @@ func (m *MockOIDC) validateCodeGrant(rw http.ResponseWriter, req *http.Request)
 	return session, true
 }
 
+func (m *MockOIDC) validateCodeChallenge(rw http.ResponseWriter, req *http.Request, session *Session) bool {
+	if session.CodeChallenge == "" || session.CodeChallengeMethod == "" {
+		return true
+	}
+
+	codeVerifier := req.Form.Get("code_verifier")
+	if codeVerifier == "" {
+		errorResponse(rw, InvalidGrant, "Invalid code verifier. Expected code but client sent none.", http.StatusUnauthorized)
+		return false
+	}
+
+	challenge, err := GenerateCodeChallenge(session.CodeChallengeMethod, codeVerifier)
+	if err != nil {
+		errorResponse(rw, InvalidRequest, "Invalid code verifier. "+err.Error(), http.StatusUnauthorized)
```

Comment:
Can we use `fmt.Sprintf` to splice on the error details instead of `+` -- that is the design pattern in the rest of the library for string management.

Reply:
I think `%v` + just the `err` as the arg right?

Reply:
Thank you for the idiomatic tip! I'm still new to Go.